.github/workflows/release-on-merge.yml

```diff
@@ -0,0 +1,64 @@
+name: Trigger release on merge
+run-name: Starting release for ${{ github.actor }} PR merge
+on:
+  pull_request:
+    types:
+      - closed
+    branches:
+      - develop
+
+jobs:
+  on_pr_merge:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Read package.json version
+        uses: actions/github-script@v6
+        id: define-package-json-version
+        with:
+          script: |
+            const { version } = require('./package.json')
+            core.info(`performing a minor release for existing version ${version}`)
+            core.setOutput('packageJsonVersion', version)
+
+      - name: Define release version
+        uses: actions/github-script@v6
+        id: define-release-version
+        with:
+          script: |
+            const { defineReleaseVersion } = require('./.github/scripts/defineVersion.js')
+            return defineReleaseVersion({core}, 'minor', undefined, "${{ steps.define-package-json-version.outputs.packageJsonVersion }}" )
+
+      - name: Bump package.json version
+        run: |
+          git config --global user.name 'Automated Release'
+          git config --global user.email 'release-automation@bitmovin.com'
+          npm version "${{ fromJson(steps.define-release-version.outputs.result).full }}"
+
+      - name: Update Changelog
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const fs = require('fs');
+            const { updateChangeLog } = require('./.github/scripts/updateChangelog.js') 
+
+            const stableVersion = '${{ fromJson(steps.define-release-version.outputs.result).full }}'.split('-')[0]
+            const releaseDate = new Date().toISOString().split('T')[0]
+
+            const data = fs.readFileSync('./CHANGELOG.md',{encoding:'utf8', flag:'r'});
+
+            core.info(`Updating ${stableVersion} with date ${releaseDate} in Changelog`);
+
+            const changelogFileContents = updateChangeLog(data, stableVersion, releaseDate);
+
+            fs.writeFileSync('./CHANGELOG.md', changelogFileContents, 'utf-8');
```

Comment:
what happens if there is no changelog entry? should we fail the release?

Reply:
Yeah, right now if there is no changelog the release will fail.

I think that should be okay, as every PR should have a changelog IMO.

Reply:
This step actually pass but later on when we commit the changes it fails as there is nothing to commit.

Reply:
👍